src/uri/path.rs

```diff
@@ -7,6 +7,18 @@ use byte_str::ByteStr;
 use convert::HttpTryFrom;
 use super::{ErrorKind, InvalidUri, InvalidUriBytes, URI_CHARS};
 
+/// Represents the segments of a URI
+#[derive(Debug)]
+pub struct PathSegments<'a>(str::Split<'a, char>);
+
+impl<'a> std::ops::Deref for PathSegments<'a> {
```

Comment:
Instead of implementing `Deref`, could you implement, could we implement `Iterator<Item = &'a str>` and `DoubleEndedIterator`? The impl would proxy to `str::Slice<'a, char>`'s implementation.

Reply:
I also think that the the CI failure is due to referencing `std` instead of `::std`.